mk/compile.mk

```diff
@@ -141,4 +141,8 @@ endef
 $(foreach f, $(srcs), $(eval $(call \
 	process_srcs,$(f),$(out-dir)/$(base-prefix)$$(basename $f).o)))
 
+# Handle generated source files, that is, files that are compiled from out-dir
+$(foreach f, $(gen-srcs), $(eval $(call \
```

Comment:
Should not we do the same for  `version.o `? Could be part of another Pull-Request not to mix feature changes (this PR is already big).